plutus-core/plutus-ir/test/PlutusIR/Compiler/Let/Test.hs

```diff
@@ -2,11 +2,44 @@ module PlutusIR.Compiler.Let.Test where
 
 import PlutusIR.Test
 
+import PlutusIR.Compiler.Let
+import PlutusIR.Properties.Typecheck
+import Test.QuickCheck
 import Test.Tasty
+import Test.Tasty.ExpectedFailure (ignoreTest)
 import Test.Tasty.Extras
+import Test.Tasty.QuickCheck
 
 test_lets :: TestTree
 test_lets = runTestNestedIn ["plutus-ir/test/PlutusIR/Compiler"] $ testNested "Let"
     [ goldenPlcFromPir pTermAsProg "letInLet"
     , goldenPlcFromPir pTermAsProg "letDep"
     ]
+
+-- FIXME
+-- | Check that a term typechecks after a
+-- `PlutusIR.Compiler.Let.compileLets` (recursive terms) pass.
+prop_TypecheckCompileLetsRecTerms :: Property
```

Comment:
I could write an arbitrary instance for [LetKind](https://github.com/input-output-hk/plutus/blob/f36d6fc5521f8740f71de109466bade38c9d10b9/plutus-core/plutus-ir/src/PlutusIR/Compiler/Let.hs#L39) too but I feel in this case it's more obviously useful to separate them. Let me know if you think otherwise.

Reply:
Yeah, in this case it actually means "do something completely different", so I think it's good to separate them.